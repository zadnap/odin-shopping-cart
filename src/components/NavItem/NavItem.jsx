import styles from './NavItem.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link, useLocation } from 'react-router-dom';

function NavItem({ to, icon, children }) {
  const location = useLocation();

  return (
    <Link
      to={to}
      className={`${styles.navItem} ${
        location.pathname === to ? styles.active : ''
      }`}
    >
      <FontAwesomeIcon icon={icon} className={styles.icon} />
      {children}
    </Link>
  );
}

export default NavItem;
