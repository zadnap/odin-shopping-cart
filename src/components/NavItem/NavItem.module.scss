@use '@/styles/variables.scss' as *;

.navItem {
  cursor: pointer;
  list-style-type: none;
  display: flex;
  color: $subtext-color;
  display: flex;
  align-items: center;
  gap: 10px;
  padding: 10px;
  transition: color ease 0.2s;

  .icon {
    width: 30px;
  }

  &.active {
    color: $text-color;
  }

  &:hover {
    background-color: $subbackground-color;
  }
}
