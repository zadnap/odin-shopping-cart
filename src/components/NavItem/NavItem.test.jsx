import { vi, describe, it, expect, beforeAll } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavItem from './NavItem';

beforeAll(() => {
  vi.mock('@fortawesome/react-fontawesome', () => ({
    FontAwesomeIcon: () => <span data-testid="fa-icon" />,
  }));
});

describe('NavItem component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <NavItem to="/movies">Movies</NavItem>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
