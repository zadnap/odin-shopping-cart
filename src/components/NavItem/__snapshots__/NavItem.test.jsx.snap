// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavItem component > should render to match snapshot 1`] = `
<div>
  <a
    class="_navItem_a6b845 "
    data-discover="true"
    href="/movies"
  >
    <span
      data-testid="fa-icon"
    />
    Movies
  </a>
</div>
`;
